filter_plugins/netdata.py

```diff
@@ -19,3 +20,15 @@ def aws_creds(self, data, type_='access'):
         if type_ == 'secret':
             return config.get('default', 'aws_secret_access_key')
         return config.get('default', 'aws_access_key_id')
+
+    def redis_roles(self, services):
+        """ Organise redis services into clusters for the redis collector """
+        tmp = dict()
+        res = []
+        for service in services:
+            tmp.setdefault(service['role'], [])
+            tmp[service['role']].append(service)
+        for idx, cluster in enumerate(list(tmp.values())):
+            for addr in cluster:
+                res.append("%s:%d:%d" % (addr['ip'], addr['port'], idx))
```

Comment:
Why are we using `idx` and not `addr['role']`? `idx` depends on the arbitrary order in the inventory, it seems like a bad idea.